pcdsdevices/attenuator.py

```diff
@@ -1101,6 +1102,101 @@ class AT2L0(FltMvInterface, PVPositionerPC, LightpathInOutMixin):
     num_in = Cpt(InternalSignal, kind='hinted')
     num_out = Cpt(InternalSignal, kind='hinted')
 
+    def _get_blade_error_attrs():
+        """Get the blade attribute names used for checking errors."""
+        for index in range(1, 20):
+            yield f"blade_{index:02d}.state.error"
+            yield f"blade_{index:02d}.state.error_id"
+            yield f"blade_{index:02d}.state.error_message"
+            yield f"blade_{index:02d}.motor.plc.err_code"
+            yield f"blade_{index:02d}.motor.user_readback"
+
+    def _check_errors(signals: SignalToValue) -> str:
+        """check for errors, return a string indicating any errors verbally"""
+        errors = []
+        # sort out .motor from .motor.plc signals
+        for sig, value in signals.items():
+            if ("motor" in sig.name) ^ ("plc" in sig.name):
+                value = int(sig.metadata["severity"])
+
+            if value not in (0, ""):
+                errors.append(f"{sig.name}: {value}")
+
+        if errors:
+            return "\n".join(["Error summary:"] + errors)
+
+        return "No Errors"
+
+    error_summary = Cpt(
+        MultiDerivedSignalRO,
+        calculate=_check_errors,
+        attrs=list(_get_blade_error_attrs()),
+        doc='summarize the errors at any time on any blade via a string',
+    )
+    set_metadata(error_summary, dict(variety='text-multiline'))
+
+    def _check_errors_bitmask(signals: SignalToValue):
+        """check for errors, return an array of binaries 1=error, 0=no error"""
+        errors = []
+        blade_errors = []
+        for sig, value in signals.items():
+            if ("motor" in sig.name) ^ ("plc" in sig.name):
+                value = int(sig.metadata["severity"])
+
+            if value not in (0, ""):
+                blade_errors.append(1)
+            else:
+                blade_errors.append(0)
+
+        step = 5
+        # first blade errors not reported in bit array
+        start_index = step
+        end_index = 2*step
+        for _ in range(1, 19):
+            error_count = sum(blade_errors[start_index:end_index])
+            errors.append(1 if error_count >= 1 else 0)
+            start_index += step
+            end_index += step
+
+        decimal_value = 0
+        for next_bit in errors:
+            decimal_value = decimal_value * 2 + next_bit
+
+        return decimal_value
+
+    """summarize errors occurring at any time on any blade via a bitmask"""
+    error_summary_bitmask = Cpt(
+        MultiDerivedSignalRO,
+        calculate=_check_errors_bitmask,
+        attrs=list(_get_blade_error_attrs()),
+        doc='summarize errors at any time on any blade via a bitmask',
+    )
+    set_metadata(error_summary_bitmask, dict(variety='bitmask', bits=18))
+
+    def clear_errors(self):
+        """wrap reset errors a function with no arguments """
```

Comment:
If this is a user-facing function, it should have a clear docstring for the user, who will be able to see it in the hutch python session. I'll make a suggestion:
```suggestion
        """Reset all attenuator errors, making the device ready to move."""
```